File: src/main/java/com/everyonewaiter/domain/auth/AuthenticationAccount.java
package com.everyonewaiter.domain.auth;

import com.everyonewaiter.domain.account.AccountPermission;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface AuthenticationAccount {

  AccountPermission permission() default AccountPermission.USER;

}

File: src/main/kotlin/com/everyonewaiter/global/annotation/AuthenticationAccount.kt
package com.everyonewaiter.global.annotation

import com.everyonewaiter.domain.account.entity.AccountPermission

@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class AuthenticationAccount(
    val permission: AccountPermission = AccountPermission.USER,
)
